### PR TITLE
chore(linter): ruff auto fix base on starbase config

### PR DESCRIPTION
**Pull request by syu-w**
All auto fix.

Depends on https://github.com/canonical/craft-parts/pull/580

### CODE REVIEW

File: pyproject.toml
Comment (lengau) [inline pyproject.toml:1]: Note to other reviewers: This is the only file that was changed other than using `ruff check --fix .`

**Review by lengau — APPROVED** [review]

File: craft_parts/callbacks.py [hunk 1]
```diff
@@ -33,7 +33,10 @@
 StepCallback = Callable[[StepInfo], bool]
 ConfigureOverlayCallback = Callable[[Path, ProjectInfo], None]
 Callback = Union[
-    FilterCallback, ExecutionCallback, StepCallback, ConfigureOverlayCallback
+    FilterCallback,
```
Comment (tigarmo) [inline craft_parts/callbacks.py:36]: did we change the desired line length?

File: craft_parts/executor/part_handler.py [hunk 2]
```diff
@@ -743,7 +772,8 @@ def _migrate_overlay_files_to_prime(self) -> None:
         for part in reversed(parts_with_overlay):
             logger.debug("migrate part %r layer to prime", part.name)
             visible_files, visible_dirs = overlays.visible_in_layer(
-                part.part_layer_dir, part.prime_dir
+                part.part_layer_dir,
```
Comment (tigarmo) [inline craft_parts/executor/part_handler.py:775]: do you know which rule is splitting lines like this one?

Comment (syu-w) [reply craft_parts/executor/part_handler.py:775]: that come from the `ruff format', which I think has no rule code

File: craft_parts/callbacks.py (hunk 1 above)
Comment (syu-w) [reply craft_parts/callbacks.py:36]: No, but the `ruff format` seems like break long lines.

**Review by cmatsuoka — CHANGES_REQUESTED** [review]
Replacing other linters with ruff looks fine, however I think we should wait for ruff formatting to get stable and use black for now (instead of always splitting argument lists).

File: craft_parts/executor/part_handler.py (hunk 2 above)
Comment (syu-w) [reply craft_parts/executor/part_handler.py:775]: OK, after some tests, this kind of format was done by `black`, not `ruff format`.

**Comment by syu-w** [comment]
Now without `COM812 # Trailing comma missing`

File: pyproject.toml [hunk 3]
```diff
@@ -138,19 +138,16 @@ ignore = [
     # Temporary ignored due to API change
     "PTH", # Use pathlib.Path
 
+    "COM812", # Trailing comma missing
```
Comment (sergiusens) [inline pyproject.toml:141]: why did you move this from the original diff?

this is still in starbase, so keeping it below seems appropriate

File: craft_parts/packages/apt_cache.py [hunk 4]
```diff
@@ -235,7 +235,7 @@ def get_installed_version(
                 return installed.version
         return None
 
-    def fetch_archives(self, download_path: Path) -> List[Tuple[str, str, Path]]:
+    def fetch_archives(self, download_path: Path) -> list[tuple[str, str, Path]]:
```
Comment (tigarmo) [inline craft_parts/packages/apt_cache.py:238]: I was going to comment on the `List` vs `list`, but apparently this file has had the `__future__.annotations` import for over two years now! Weird

**Review by tigarmo — APPROVED** [review]
thanks!

File: tests/conftest.py [hunk 5]
```diff
@@ -211,13 +210,13 @@ def create_dependency_fixture(
...
 
-    yield create_dependency_fixture
+    return create_dependency_fixture
 
 
 ChmodCall = collections.namedtuple("ChmodCall", ["owner", "group", "kwargs"])
 
 
-@pytest.fixture
+@pytest.fixture()
```
Comment (cmatsuoka) [inline tests/conftest.py:219]: Any special reason not to use `dict` in the line below?

File: tests/integration/executor/test_callbacks.py [hunk 6]
```diff
@@ -132,7 +131,7 @@ def test_prologue_callback(tmpdir, capfd):
 
 
 def _my_step_callback(info: StepInfo) -> bool:
-    msg = getattr(info, "message")
+    msg = info.message
```
Comment (cmatsuoka) [inline tests/integration/executor/test_callbacks.py:134]: We can remove the assignment and print `info.message` directly.

File: tests/integration/executor/test_callbacks.py [hunk 7]
```diff
@@ -330,7 +329,7 @@ def test_invalid_update_callback_post(tmpdir, step):
 
 
 def _my_exec_callback(info: ProjectInfo) -> None:
-    print(f"{getattr(info, 'message')}")
+    print(f"{info.message}")
```
Comment (cmatsuoka) [inline tests/integration/executor/test_callbacks.py:332]: Any special reason not to print directly instead of using the f-string?

File: tests/integration/plugins/test_python.py [hunk 8]
```diff
@@ -20,12 +20,11 @@
...
 from typing import Optional
 
+import craft_parts.plugins.plugins
 import pytest
 import yaml
-from overrides import override
-
-import craft_parts.plugins.plugins
 from craft_parts import LifecycleManager, Step, errors, plugins
+from overrides import override
```
Comment (cmatsuoka) [inline tests/integration/plugins/test_python.py:27]: Interesting order swap here, shouldn't third-party imports be in the middle block and project-local imports in the bottom block?

File: tests/unit/plugins/test_ant_plugin.py [hunk 9]
```diff
@@ -15,13 +15,12 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import pytest
-from pydantic import ValidationError
-
 from craft_parts import Part, PartInfo, ProjectInfo, errors
 from craft_parts.plugins.ant_plugin import AntPlugin
+from pydantic import ValidationError
```
Comment (cmatsuoka) [inline tests/unit/plugins/test_ant_plugin.py:20]: Same third-party vs local reordering here. Is isort happy with that? Otherwise it could be a ruff bug or configuration.

File: tests/unit/plugins/test_autotools_plugin.py [hunk 10]
```diff
@@ -17,11 +17,10 @@
 from pathlib import Path
 
 import pytest
-from pydantic import ValidationError
-
 from craft_parts.infos import PartInfo, ProjectInfo
 from craft_parts.parts import Part
 from craft_parts.plugins.autotools_plugin import AutotoolsPlugin
+from pydantic import ValidationError
```
Comment (cmatsuoka) [inline tests/unit/plugins/test_autotools_plugin.py:23]: Ditto.

File: tests/unit/utils/test_file_utils.py [hunk 11]
```diff
@@ -129,7 +128,7 @@ def test_link_file_ioerror(self, mocker):
 
         def link_and_ioerror(a, b, **kwargs):  # pylint: disable=unused-argument
             orig_link(a, b)
-            raise IOError()
+            raise OSError()
```
Comment (cmatsuoka) [inline tests/unit/utils/test_file_utils.py:131]: Maybe we could rename the function to `link_and_oserror`.

**Review by cmatsuoka — COMMENTED** [review]
Nice! Added a few questions and nitpicks but overall it looks pretty good!

**Review by cmatsuoka — APPROVED** [review]

File: tests/integration/plugins/test_python.py (hunk 8 above)
Comment (syu-w) [reply tests/integration/plugins/test_python.py:27]: I guess the test cases do not count as project-local imports

File: tests/unit/plugins/test_ant_plugin.py (hunk 9 above)
Comment (syu-w) [reply tests/unit/plugins/test_ant_plugin.py:20]: isort moved it to the top

**Comment by syu-w** [comment]
There could be some current ignored rules make the import order changed. Let me see if following PRs will fix that or not.

File: pyproject.toml (hunk 3 above)
Comment (syu-w) [reply pyproject.toml:141]: I will remove the ignored rules below in the following PRs that fix them, so I put this one on top that should not be removed.

File: craft_parts/packages/apt_cache.py (hunk 4 above)
Comment (syu-w) [reply craft_parts/packages/apt_cache.py]: I removed `__future__.annotations` for now to keep the style the same with other files.

File: tests/conftest.py (hunk 5 above)
Comment (syu-w) [reply tests/conftest.py:219]: Just keep the old style for now
